site/app/models/DockerUI.php

```diff
@@ -0,0 +1,501 @@
+<?php
+
+namespace app\models;
+
+use app\libraries\Core;
+use app\libraries\DateUtils;
+use app\libraries\Utils;
+use app\libraries\FileUtils;
+use app\libraries\Logger;
+//re-using the php parseError, this is typically used internally but for now using for errors related to parsing logs
+use ParseError;
+use Exception;
+
+/**
+ * Simple class to represent a worker machine
+ */
+class Worker {
+    /** The name of this worker */
+    public string $name;
+    /** number of workers this machine has */
+    public int $num_autograding_workers;
+    /** What capabilities this worker has */
+    public array $capabilities;
+    /** Is this woker enabled or not */
+    public bool $is_enabled;
+    /** Set after parsing logs if this machine could not be updated */
+    public bool $failed_to_update;
+    /** Docker version on the machine */
+    public string $docker_version;
+    /** OS of the machine */
+    public string $os;
+    /** System information parsed from sysinfo logs */
+    public ?WorkerSystemInformation $system_information;
+    /** Create a new Worker object */
+    public function __construct(string $name, int $num_autograding_workers, array $capabilities, bool $is_enabled) {
+        $this->name = $name;
+        $this->num_autograding_workers = $num_autograding_workers;
+        $this->capabilities = $capabilities;
+        $this->is_enabled = $is_enabled;
+        //these pieces of information are known later in the parsing
+        $this->failed_to_update = false;
+        $this->system_information = null;
+        // this information is taken from the docker logs parsed earlier and associated after the worker obj is created
+        $this->docker_version = "Unknown";
+        $this->os = "Unknown";
+    }
+}
+
+/**
+ * Store information about a worker system information, this comes from system log information 
+ * instead of configuration
+ */
+class WorkerSystemInformation {
+    /** Name of worker machine this information is associated with */
+    public string $associated_worker; 
+    /** State of worker_service */
+    public string $worker_service;
+    /** State of worker_service */
+    public ?string $shipper_service;
+    /** State of worker_service */
+    public ?string $daemon_service;
+    /** Disk usage in percetange */
+    public string $disk_usage;
+    /** String with a unix load average
+     * See https://www.gnu.org/software/coreutils/manual/html_node/uptime-invocation.html#uptime-invocation 
+     * For understanding the load numbers
+     */
+    public string $load;
+    /** operating system of the worker */
+    public string $os;
+
+    /** Create a new WorkerSystemInformation object */
+    public function __construct(string $associated_worker, string $worker_service, string $shipper_service, string $daemon_service, 
+        string $disk_usage, string $load) {
```

Comment:
If you want to wrap this line, it might be better to put each parameter on a separate line